More Mweep

Yomitht opened the pull request:
Adjusted the "King Zora Speed" timesaver enhancement to allow for slowing the animation down.

Feature Request #4186

Yomitht committed:
More Mweep

Adjusted the "King Zora Speed" timesaver enhancement to allow for slowing the animation down.

inspectredc commented on soh/soh/SohMenuBar.cpp:
```diff
@@ -570,7 +570,8 @@ void DrawEnhancementsMenu() {
                 UIWidgets::PaddedEnhancementCheckbox("Skip Text", CVAR_ENHANCEMENT("SkipText"), false, true);
                 UIWidgets::Tooltip("Holding down B skips text");
-                UIWidgets::PaddedEnhancementSliderInt("King Zora Speed: %dx", "##MWEEPSPEED", CVAR_ENHANCEMENT("MweepSpeed"), 1, 5, "", 1, true, false, true);
+                //UIWidgets::PaddedEnhancementSliderInt("King Zora Speed: %dx", "##MWEEPSPEED", CVAR_ENHANCEMENT("MweepSpeed"), 1, 5, "", 1, true, false, true);
```
Do you mind just removing the line here rather than commenting

Yomitht replied:
Yes, I'll get right on that. Thanks for reminding me.

Yomitht replied:
Done

inspectredc commented on soh/soh/SohMenuBar.cpp:
```diff
@@ -570,7 +570,8 @@ void DrawEnhancementsMenu() {
                 UIWidgets::Tooltip("Holding down B skips text");
-                UIWidgets::PaddedEnhancementSliderInt("King Zora Speed: %dx", "##MWEEPSPEED", CVAR_ENHANCEMENT("MweepSpeed"), 1, 5, "", 1, true, false, true);
+                //UIWidgets::PaddedEnhancementSliderInt("King Zora Speed: %dx", "##MWEEPSPEED", CVAR_ENHANCEMENT("MweepSpeed"), 1, 5, "", 1, true, false, true);
+                UIWidgets::PaddedEnhancementSliderFloat("King Zora Speed: %.2fx %%", "##MWEEPSPEED", CVAR_ENHANCEMENT("MweepSpeed"), 0.1f, 5.0f, "", 1.0f, true, false, true);
```
I believe these `%%`s on other sliders just add a percentage sign, and we just want to stick with the multiplier sign for this

```suggestion
                UIWidgets::PaddedEnhancementSliderFloat("King Zora Speed: %.2fx", "##MWEEPSPEED", CVAR_ENHANCEMENT("MweepSpeed"), 0.1f, 5.0f, "", 1.0f, true, false, true);
```

Yomitht replied:
Yes, that's what I gathered. I switched it over to a percentage display because I was having trouble getting it to display properly, but I'll see if I can set it back to a multiplier.

Yomitht replied:
Done

inspectredc reviewed (commented):
Awesome, much needed :D! Left a few tidy up comments

Yomitht committed:
[Fixed] Crashing when applying preset settings

Adjusted presets.h to expect a float instead of an integer for the "MweepSpeed" timesaver enhancement setting.

Yomitht committed:
Cleaning some things up

Deleted an unnecessary line, and switched the "King Zora Speed" option back to multiplier display instead of a percentage.

aMannus reviewed (commented):
As a small note, we've moved away from float sliders (and later convert the integer into a float instead) because the imgui float sliders are inaccurate and nearly impossible to reset to a default state unless programmatically set back to whatever value. 

Not really sure how you'd handle that for fractions below 1, but there was good reason for not using float sliders before.

Yomitht commented:
> As a small note, we've moved away from float sliders (and later convert the integer into a float instead) because the imgui float sliders are inaccurate and nearly impossible to reset to a default state unless programmatically set back to whatever value.
> 
> Not really sure how you'd handle that for fractions below 1, but there was good reason for not using float sliders before.

Yeah, I noticed that. Do you know if we have a prebuilt UI element that lets you just enter a value?

Yomitht committed:
Update SohMenuBar.cpp

Switched back to multiplier display

inspectredc reviewed (approved):
LGTM! Can't pull right now to double check everything but doubt there's any issues

Yomitht committed:
Merge branch 'HarbourMasters:develop' into King-Zora-Slowness

Yomitht commented:
Closes #4186

Yomitht commented:
Closed #4186

Malkierian committed:
Merge branch 'develop' into King-Zora-Slowness
